src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -14,6 +14,18 @@
             [varity.ref-gene :as rg]
             [varity.vcf-to-hgvs.common :refer [diff-bases] :as common]))
 
+(defn- overlap-exon-intron-boundary?
+  [exon-ranges pos ref alt]
+  (let [nref (count ref)
+        nalt (count alt)]
+    (and (not (= 1 nref nalt))
+         (not= 1 (count exon-ranges))
+         (some (fn [[s e]]
+                 (and (not= s e)
+                      (or (and (< pos s) (<= s (+ pos nref -1)))
+                          (and (<= pos e) (< e (+ pos nref -1))))))
+               exon-ranges))))
```

Comment:
I split exon/intron boundary decision function from `alt-exon-range` and use it in `alt-exon-ranges` and `read-sequence-info`.